ReserveRoom.java

```diff
@@ -3,106 +3,93 @@
 import java.util.Scanner;
 
 public class ReserveRoom {
-    private List<Room> rooms; // List of rooms
-    private List<String> reservations; // List of room reservations
+    private List<Room> rooms;         
+    private List<String> reservations;  
 
-    // Constructor initializes the list of rooms and reservations
     public ReserveRoom(List<Room> rooms) {
         this.rooms = rooms;
         this.reservations = new ArrayList<>();
     }
+    private User promptUserDetails(Scanner scanner) {
+        System.out.print("Enter your name: ");
+        String name = scanner.nextLine();
+        System.out.print("Enter your email: ");
+        String email = scanner.nextLine();
```

Comment:
// TO DO: Trim whitespace from all user inputs to avoid storing unintended spaces. (Performance & Security)
```suggestion
        String email = scanner.nextLine().trim();  // Trims whitespace
```